pkg/manager/manager_test.go

```diff
@@ -469,6 +469,40 @@ var _ = Describe("manger.Manager", func() {
 				ok := metrics.Registry.Unregister(one)
 				Expect(ok).To(BeTrue())
 			})
+
+			It("should serve extra endpoints", func(done Done) {
+				opts.MetricsBindAddress = ":0"
+				m, err := New(cfg, opts)
+				Expect(err).NotTo(HaveOccurred())
+
+				err = m.AddMetricsExtraHandler("/debug", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+					_, _ = w.Write([]byte("Some debug info"))
+				}))
+				Expect(err).NotTo(HaveOccurred())
+
+				// Should error when we add another extra endpoint on the already registered path.
+				err = m.AddMetricsExtraHandler("/debug", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+					_, _ = w.Write([]byte("Another debug info"))
+				}))
+				Expect(err).To(HaveOccurred())
+
+				s := make(chan struct{})
+				defer close(s)
+				go func() {
+					defer GinkgoRecover()
+					Expect(m.Start(s)).NotTo(HaveOccurred())
```

Comment:
Doesn't the start race with the the Get request we do later on?

Reply:
I've done this in a similar way as in other tests where HTTP requests are involved. It seems to work, but I see no strong argument why this could not race potentially, maybe I'm missing something.

Reply:
hm okay, interesting. Well, we can still look at it if we actually see it flake.